src/services/jobs/StatisticsService.java

```diff
@@ -36,20 +43,20 @@ public StatisticsService() {
         this(HibernateUtil.getSessionFactory());
     }
 
-    public Map<String,Integer> getAreaData(){
+    public Map<String,Long> getAreaData(){
        Session session = sessionFactory.getCurrentSession();
        Transaction tx = session.beginTransaction();
-       Map<String,Integer> data;
+       Map<String,Long> data;
        try{
             List<JobArea> areas = session.createCriteria(JobArea.class).list();
             data = new HashMap<>();
             for(JobArea area : areas){
                 data.put(
                         area.getName(), 
-                        (Integer) session.createCriteria(Job.class).
+                        (Long) session.createCriteria(Job.class).
+                                createAlias("jobArea", "jobArea").
                                 setProjection(Projections.rowCount()).
-                                setFetchMode("jobfeature", FetchMode.JOIN).
-                                setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY).
+                                add(Restrictions.like("jobArea.name", area.getName())).
                                 uniqueResult()
```

Comment:
Would it be possible to fetch all these entities at once? You're performing N + 1 selects (one for JobAreas and N for Jobs per area). You could, for example, create a query that fetches rowCount() grouped by jobArea.id (do note that you should use jobArea.id rather than jobArea.name because name might be duplicated)